dallinger/mturk.py

```
@@ -182,104 +248,112 @@ def get_qualification_type_by_name(self, name):
-            new_score = current_score + 1
-            self.update_qualification_score(qtype_id, worker_id, new_score)
-        else:
-            self.assign_qualification(
-                qtype_id, worker_id, new_score, notify)
+        self.update_qualification_score(qtype_id, worker_id, new_score)
 
         return {
             'qtype': result['qtype'],
             'score': new_score
         }
 
-    def update_qualification_score(self, qualification_id, worker_id, score):
-        """Score a worker for a specific qualification"""
-        return self._is_ok(self.mturk.update_qualification_score(
-            qualification_id,
-            worker_id,
-            score,
+    def revoke_qualification(self, qualification_id, worker_id, reason=''):
+        return self._is_ok(self.mturk.disassociate_qualification_from_worker(
+            QualificationTypeId=qualification_id,
+            WorkerId=worker_id,
+            Reason=reason
         ))
 
+    def get_qualification_score(self, qualification_id, worker_id):
+        """Return a worker's qualification score as an iteger.
+        """
+        try:
+            response = self.mturk.get_qualification_score(
+                QualificationTypeId=qualification_id,
+                WorkerId=worker_id
+            )
+        except ClientError as ex:
+            raise MTurkServiceException(ex.message)
+        return response['Qualification']['IntegerValue']
+
+    def get_current_qualification_score(self, name, worker_id):
+        """Return the current score for a worker, on a qualification with the
+        provided name.
+        """
+        qtype = self.get_qualification_type_by_name(name)
+        if qtype is None:
+            raise QualificationNotFoundException(
+                'No Qualification exists with name "{}"'.format(name)
+            )
+        try:
+            score = self.get_qualification_score(qtype['id'], worker_id)
+        except MTurkServiceException:
+            score = None
```

Do we really want to swallow all MTurk exceptions?